plugins/inputs/logfile/tailersrc.go

```diff
@@ -306,6 +307,13 @@ func (ts *tailerSrc) runSaveState() {
 				continue
 			}
 			lastSavedOffset = offset
+		case <-ts.tailer.FileDeletedCh:
```

Comment:
When a file is deleted, is this case guaranteed to happen before `case <-ts.done:`?

Reply:
Yep.
```go
			err := tail.waitForChanges()
			if err != nil {
				if err == ErrDeletedNotReOpen {
					close(tail.FileDeletedCh)
```
The tail process waits for changes, sees that the file was deleted, and then I close the channel

Reply:
Curious question: If there is anything wrong when reading the remaining lines in log files, will closing the channels would be a good idea while we want to maintain the offset?

Reply:
Can you clarify what you mean by "no matter the error"? The only time that we get an `ErrDeletedNotReOpen` is when the log file gets deleted.

Reply:
Its the afterwards of `closing the channel` in [these lines](https://github.com/aws/amazon-cloudwatch-agent/pull/457/files#diff-b4cbda30ad21153305e4658dca8a5844580e52b12544b5147237a3315d185284R393-R398). Should we only close the channel when we send these lines successfully since we want to keep the offset for unsuccessful lines? (or maybe I am not understand thing here)